ci/ray_ci/test_utils.py

```diff
@@ -34,5 +41,48 @@ def _mock_subprocess_run(
         docker_login("docker_ecr")
 
 
+def _make_test(name: str, state: str, team: str) -> Test:
+    return Test(
+        {
+            "name": name,
+            "state": state,
+            "team": team,
+        }
+    )
+
+
+@mock.patch("ray_release.test.Test.gen_from_s3")
+def test_get_flaky_test_names(mock_gen_from_s3):
+    mock_gen_from_s3.side_effect = (
+        [
+            _make_test("darwin://test_1", "flaky", "core"),
+            _make_test("darwin://test_2", "flaky", "ci"),
+        ],
+    )
```

Comment:
could you add some test data that is not started with `darwin:` and of other state?

Reply:
done!